src/data/finsupp.lean

```diff
@@ -206,6 +206,65 @@ finsupp.ext $ λ a', show f (ite _ _ _) = ite _ _ _, by split_ifs; [refl, exact
 
 end map_range
 
+section emb_domain
```

Comment:
Coincidence: I had just written this section together with Fabian. Although we extended from a set, instead of an embedding. This is more general, and we can easily apply it in our use case.